### PR TITLE
Simplify trade hierarchy

### PR DESCRIPTION
**Pull request by jodastephen**
Remove `Trade` from strata-basics.
Rename `FinanceTrade` to `Trade` in strata-product.

### CODE REVIEW

File: modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java [hunk 1]
```diff
@@ -33,12 +33,11 @@ public void compose() {
     CalculationFunctions fns2 = CalculationFunctions.of(fn1b, fn2);
     CalculationFunctions composed = fns1.composedWith(fns2);
 
-    assertThat(composed.getFunction(new Target1())).isEqualTo(fn1a);
-    assertThat(composed.getFunction(new Target2())).isEqualTo(fn2);
+    assertEquals(composed.getFunction(new Target1()), fn1a);
```
Comment (cjkent) [inline modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java]: Why change assertions? I prefer AssertJ but don't think it's worth switching every test I touch.

File: modules/report/src/main/java/com/opengamma/strata/report/framework/expression/TradeTokenEvaluator.java [hunk 2]
```diff
@@ -58,19 +53,15 @@ public EvaluationResult evaluate(Trade trade, String firstToken, List<String> re
...
-      if (tradeInfoPropertyName.isPresent()) {
-        Object propertyValue = ftrade.getInfo().property(tradeInfoPropertyName.get()).get();
+    if (tradeInfoPropertyName.isPresent()) {
+      Object propertyValue = trade.getInfo().property(tradeInfoPropertyName.get()).get();
 
-        return propertyValue != null ?
-            EvaluationResult.success(propertyValue, remainingTokens) :
-            EvaluationResult.failure("Property '{}' not set", firstToken);
-      }
+      return propertyValue != null ? EvaluationResult.success(propertyValue, remainingTokens)
```
Comment (cjkent) [inline modules/report/src/main/java/com/opengamma/strata/report/framework/expression/TradeTokenEvaluator.java]: The old formatting was easier to understand.

File: modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java (hunk 1 above)
Comment (jodastephen) [inline modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java]: Code failed in Eclipse with assertThat. Given it was the only use in the test it was quicker for me to change than think about it.

File: modules/report/src/main/java/com/opengamma/strata/report/framework/expression/TradeTokenEvaluator.java (hunk 2 above)
Comment (jodastephen) [inline modules/report/src/main/java/com/opengamma/strata/report/framework/expression/TradeTokenEvaluator.java]: Code was auto-formatted without me noticing...

File: modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java (hunk 1 above)
Comment (cjkent) [inline modules/calc/src/test/java/com/opengamma/strata/calc/runner/CompositeCalculationFunctionsTest.java]: https://www.jetbrains.com/idea/download/ ;)

**Pull request merged**